api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -211,11 +215,35 @@ private Obs saveObsNotDirty(Obs obs, String changeMessage) {
 	}
 
 	private Obs saveNewOrVoidedObs(Obs obs, String changeMessage) {
+		setObsReferenceRange(obs);
 		Obs ret = dao.saveObs(obs);
 		saveObsGroup(ret,changeMessage);
 		return ret;
 	}
 
+	private void setObsReferenceRange(Obs obs) {
+		Concept concept = obs.getConcept();
+		ConceptService conceptService = Context.getConceptService();
+
+		if (concept != null && concept.getDatatype().isNumeric()) {
+
+			Optional<ConceptReferenceRange> conceptReferenceRange = conceptService.getConceptReferenceRangeByConceptId(concept.getId());
```

Comment:
Because we can have multiple ranges for a concept numeric, how are you determining which one to assign to the obs?

Reply:
What's your suggestion here?

Reply:
The criteria evaluation determines which range applies for a given person.

Reply:
Do you get slack notifications?

Reply:
Update: this has been discussed on slack.